pkg/flow/limiter_test.go

```diff
@@ -59,23 +60,48 @@ func TestCapacityLimiter_Drop(t *testing.T) {
 	// BUT not a single extra element
 	select {
 	case elem := <-pipeOut:
-		assert.Failf(t, "unexpected element", "%#v", elem)
+		var first *model.Record
+		if len(elem) > 0 {
+			first = elem[0]
+		}
+		assert.Failf(t, "unexpected element", "size: %d, first: %#v", len(elem), first)
 	default:
 		// ok!
 	}
 }
 
-func capacityLimiterPipe() (in chan<- []*model.Record, out <-chan []*model.Record) {
+func capacityLimiterPipe() (chan<- []*model.Record, <-chan []*model.Record) {
 	inCh, outCh := make(chan []*model.Record), make(chan []*model.Record)
 
 	init := node.AsInit(func(initOut chan<- []*model.Record) {
 		for i := range inCh {
+			// fmt.Printf("in: %s\n", i[0].Interfaces[0].Interface)
```

Comment:
```suggestion
```

Reply:
I wanted to keep it because I'm not 100% sure it's a definitive fix (it's always the problem with race conditions...) and in case I see it again, it'll be faster to debug again

Reply:
anyway, ok I can remove them, hopefully the fix is definitive :-)